File: src/main/java/com/example/NewsAI/services/LikeService.java
package com.example.NewsAI.services;

import com.example.NewsAI.entities.Like;

import java.util.Map;

public interface LikeService {

    Like likeArticle(Like like, String userEmail);

    void unlike(Long id);

    boolean checkIfLiked(String userEmail, Long articleId);

    long getLikeCount(Long articleId);

    Map<String, Object> toggleLike(Long articleId, String userEmail);
}

File: src/main/java/com/example/IslyNews/frontend/src/app/core/services/like.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Like, CreateLikeRequest } from '../models/like.model';

@Injectable({ providedIn: 'root' })
export class LikeService {
  private readonly API = 'http://localhost:8082/likes';

  constructor(private http: HttpClient) {}

  like(req: CreateLikeRequest): Observable<Like> {
    return this.http.post<Like>(this.API, req);
  }

  unlike(id: number): Observable<void> {
    return this.http.delete<void>(`${this.API}/${id}`);
  }

  checkIfLiked(articleId: number): Observable<boolean> {
    return this.http.get<boolean>(`${this.API}/check/${articleId}`);
  }

  toggleLike(articleId: number): Observable<{ liked: boolean; message: string }> {
    return this.http.post<{ liked: boolean; message: string }>(`${this.API}/toggle/${articleId}`, {});
  }
}
